sdk/core/Azure.Core/src/Pipeline/HttpPipelineTransport.cs

```diff
@@ -28,5 +29,46 @@ public abstract class HttpPipelineTransport
         /// </summary>
         /// <returns></returns>
         public abstract Request CreateRequest();
+
+        /// <summary>
+        /// Creates the default <see cref="HttpPipelineTransport"/> based on the current environment and configuration.
+        /// </summary>
+        /// <param name="options"><see cref="HttpPipelineTransportOptions"/> that affect how the transport is configured.</param>
+        /// <returns></returns>
+        internal static HttpPipelineTransport Create(HttpPipelineTransportOptions? options = null)
+        {
+#if NETFRAMEWORK
+            if (!AppContextSwitchHelper.GetConfigValue(
+                "Azure.Core.Pipeline.DisableHttpWebRequestTransport",
+                "AZURE_CORE_DISABLE_HTTPWEBREQUESTTRANSPORT"))
+            {
+                return options switch
+                {
+                    null => HttpWebRequestTransport.Shared,
+                    _ => new HttpWebRequestTransport(options)
+                };
+            }
+#endif
+            return options switch
+            {
+                null => HttpClientTransport.Shared,
+                _ => new HttpClientTransport(options)
+            };
+        }
+
+        /// <summary>
+        /// Disposes the underlying transport.
+        /// </summary>
+        internal void Dispose()
```

Comment:
I wonder if we should do the full Dispose(bool disposing) implementation here.

Reply:
Or just have subclasses implement IDisposable on their own and have a conditional cast in Pipeline.Dipose(),

Reply:
what's the advantage to doing this?

Reply:
Hm, why not implement IDisposable on individual transports?

Reply:
We'd have to expose a public Dispose method on the transport rather than on the DisposableTransport wrapper.

Reply:
We don't have disposableTransport, do we? Only DisposablePipeline.

Reply:
> We don't have disposableTransport, do we? Only DisposablePipeline.

Yes, that is what I meant. But the same comment applies as far as making the individual transports disposable.

Reply:
Yeah, we'll have to implement the interface but I don't think it's bad. Customers don't have access to transport instances most of the time. And when they create their own transports disposing them is the right thing.